compiler/x/env/OMRCPU.cpp

```diff
@@ -144,5 +150,8 @@ OMR::X86::CPU::getX86OSSupportsSSE2(TR::Compilation *comp)
 bool
 OMR::X86::CPU::getX86SupportsTM(TR::Compilation *comp)
    {
-   return false;
+   uint32_t flags8 = self()->getX86ProcessorFeatureFlags8(comp);
```

Comment:
I'm not sure if this is the right query due to the possibility that some Haswell chips could be running with out the microcode update that disabled this feature due the hardware having implementation bugs. For true safety this check probably needs to ensure the processor is of a sufficiently new model as well as having the feature flag set to avoid weird problems from bugs in earlier hardware which may not have been patched.